File: .pub-cache/hosted/pub.dartlang.org/camera_android-0.9.8+3/android/src/test/java/io/flutter/plugins/camera/types/FlashModeTest.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camera.types;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class FlashModeTest {

  @Test
  public void getValueForString_returnsCorrectValues() {
    assertEquals(
        "Returns FlashMode.off for 'off'", FlashMode.getValueForString("off"), FlashMode.off);
    assertEquals(
        "Returns FlashMode.auto for 'auto'", FlashMode.getValueForString("auto"), FlashMode.auto);
    assertEquals(
        "Returns FlashMode.always for 'always'",
        FlashMode.getValueForString("always"),
        FlashMode.always);
    assertEquals(
        "Returns FlashMode.torch for 'torch'",
        FlashMode.getValueForString("torch"),
        FlashMode.torch);
  }

  @Test
  public void getValueForString_returnsNullForNonexistantValue() {
    assertEquals(
        "Returns null for 'nonexistant'", FlashMode.getValueForString("nonexistant"), null);
  }

  @Test
  public void toString_returnsCorrectValue() {
    assertEquals("Returns 'off' for FlashMode.off", FlashMode.off.toString(), "off");
    assertEquals("Returns 'auto' for FlashMode.auto", FlashMode.auto.toString(), "auto");
    assertEquals("Returns 'always' for FlashMode.always", FlashMode.always.toString(), "always");
    assertEquals("Returns 'torch' for FlashMode.torch", FlashMode.torch.toString(), "torch");
  }
}

File: .pub-cache/hosted/pub.dartlang.org/camera_platform_interface-2.2.0/test/types/flash_mode_test.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:camera_platform_interface/camera_platform_interface.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('FlashMode should contain 4 options', () {
    const List<FlashMode> values = FlashMode.values;

    expect(values.length, 4);
  });

  test('FlashMode enum should have items in correct index', () {
    const List<FlashMode> values = FlashMode.values;

    expect(values[0], FlashMode.off);
    expect(values[1], FlashMode.auto);
    expect(values[2], FlashMode.always);
    expect(values[3], FlashMode.torch);
  });
}
